src/main/scala/com/databricks/labs/overwatch/pipeline/SilverTransforms.scala

```diff
@@ -1314,4 +1314,119 @@ trait SilverTransforms extends SparkSessionWrapper {
 
   }
 
+  private def warehouseBase(auditRawDF: DataFrame): DataFrame = {
+    val warehouse_id_gen_w = Window.partitionBy('organization_id, 'warehouse_name)
+      .orderBy('timestamp).rowsBetween(Window.currentRow, 1000)
+    val warehouse_name_gen_w = Window.partitionBy('organization_id, 'warehouse_id)
+      .orderBy('timestamp).rowsBetween(Window.currentRow, 1000)
+    val warehouse_id_gen = first('warehouse_id, true).over(warehouse_id_gen_w)
+    val warehouse_name_gen = first('warehouse_name, true).over(warehouse_name_gen_w)
```

Comment:
This logic appears unnecessary and incorrect. Warehouse_id is a key -- should we really be filling like this? Are there really actions in scope that result in no warehouse id?

Warehouse_name -- I see that you're trying to fill forward -- that makes sense -- but please use the appropriate fillForward commands -- this is not accurate because your getting the first non_null warehouse name and it's not even ordered. Additionally, this likely won't help you at all since you're not pulling in lagging history so you likely won't have any previous data in many records. Please reference the other fillForward examples and consider pulling in an appropriate amount of historical if you intend to fill. We can discuss further if you like.

Reply:
Yes we have found null warehouse ids. Thats why we are filling it forward